### PR TITLE
PUBDEV-9073: parallelize maxrsweep

### PR DESCRIPTION
**Pull request by wendycwong**
This PR completes the work in JIRA: https://h2oai.atlassian.net/browse/PUBDEV-9073

In general, the CPM is swept each time we add a predictor to the predictor subset.  The CPM is stored normally as a double[][] array.  Hence, during each sweeping action, only one node is involved.  To parallelize this operation, I have chosen to make CPM into a frame and conduct the sweeping per chunk.  This is described in the doc attached to the JIRA.

Java/Python/R tests have been added to make sure that the multinode_mode is working correctly.

I need to add a timing tests to make sure the the multinode mode does not slow down the operation.

### CODE REVIEW

File: h2o-algos/src/main/java/hex/schemas/ModelSelectionV3.java [hunk 1]
```diff
@@ -114,6 +115,11 @@ public static final class ModelSelectionParametersV3 extends ModelParametersSche
                 level = API.Level.critical)
         public boolean lambda_search;
 
+        @API(help = "For maxrsweep only.  If enabled will attempt to perform sweeping action using multiple nodes in" +
```
Comment (tomasfryda) [inline h2o-algos/src/main/java/hex/schemas/ModelSelectionV3.java:118]:
```suggestion
        @API(help = "For maxrsweep only.  If enabled will attempt to perform sweeping action using multiple nodes in " +
```
missing space after `in` causes `inthe` in R and python doc-strings

File: h2o-r/h2o-package/R/sdt.R
Comment (tomasfryda) [inline h2o-r/h2o-package/R/sdt.R:1]: Remove this. Single decision tree should not be part of this PR.

File: h2o-r/tests/testdir_algos/glm/testthat-problems.rds
Comment (tomasfryda) [inline h2o-r/tests/testdir_algos/glm/testthat-problems.rds:1]: I don't think this should be committed.

File: h2o-py/tests/testdir_algos/gam/pyunit_sf_monotone_splines.py
Comment (tomasfryda) [inline h2o-py/tests/testdir_algos/gam/pyunit_sf_monotone_splines.py:1]: Should this be a part of this PR?

**Review by tomasfryda — CHANGES_REQUESTED** [review]
The code looks good but I have some concerns about files included in this PR.

File: h2o-algos/src/main/java/hex/modelselection/ModelSelectionModel.java [hunk 2]
```diff
@@ -84,8 +84,9 @@ public static class ModelSelectionParameters extends Model.Parameters {
         public double _obj_reg = -1.0;
         public double[] _lambda = new double[]{0.0};
         public boolean _use_all_factor_levels = false;
-        public boolean _build_glm_model = true;
+        public boolean _build_glm_model = false;
```
[the remark this answers is not in the document]
Comment (wendycwong) [reply h2o-algos/src/main/java/hex/modelselection/ModelSelectionModel.java:87]: Tomas:  This parameter is just for our one customer.  If this parameter is set to true, it will return a GLM model associated with the best predictor subset.  If it is set to false, it will not return the GLM model, it will just return the predictor subset names and the coefficients associated with the predictors.  

Talked to them and they don't care about getting the GLM model but rather wants things to run fast.  Hence, the change.

File: h2o-algos/src/main/java/hex/modelselection/ModelSelection.java [hunk 3]
```diff
@@ -537,9 +575,9 @@ public static SweepModel replacement(List<Integer> currSubsetIndices, List<Integ
                 ArrayList<Integer> oneLessSubset = new ArrayList<>(currSubsetIndices);
                 int removedPred = oneLessSubset.remove(index);
                 validSubset.removeAll(oneLessSubset);
-                currModel._predSubset = oneLessSubset.stream().mapToInt(Integer::intValue).toArray();
-                tempModel = forwardStepR(currSubsetIndices, validSubset, usedCombos, predIndices, currCPM,
-                        predictorIndex2CPMIndices, currModel, hasIntercept, errorVarianceMin, index, currCPM);
+                currModel._predSubset = oneLessSubset.stream().mapToInt(x -> x).toArray();
```
[the remark this answers is not in the document]
Comment (wendycwong) [reply h2o-algos/src/main/java/hex/modelselection/ModelSelection.java:578]: It feels redundant since mapToInt will change Integer to int.

File: h2o-r/h2o-package/R/sdt.R
Comment (wendycwong) [reply h2o-r/h2o-package/R/sdt.R:1]: done.

File: h2o-py/tests/testdir_algos/gam/pyunit_sf_monotone_splines.py
Comment (wendycwong) [reply h2o-py/tests/testdir_algos/gam/pyunit_sf_monotone_splines.py:1]: nope.

File: h2o-r/tests/testdir_algos/glm/testthat-problems.rds
Comment (wendycwong) [reply h2o-r/tests/testdir_algos/glm/testthat-problems.rds:1]: Removed

File: h2o-py/h2o/estimators/model_selection.py [hunk 4]
```diff
@@ -325,6 +326,10 @@ def __init__(self,
                excluded in the dataset.
                Defaults to ``None``.
         :type influence: Literal["dfbetas"], optional
+        :param multinode_mode: For maxrsweep only.  If enabled will attempt to perform sweeping action using multiple
+               nodes inthe cluster.  Defaults to false.
```
Comment (tomasfryda) [inline h2o-py/h2o/estimators/model_selection.py:330]:
```
:param multinode_mode: For maxrsweep only.  If enabled will attempt to perform sweeping action using multiple
               nodes __inthe__ cluster.  Defaults to false.
```
needs regenerating the client code before merging. `ModelSelectionV3` contains the correct version.

**Review by tomasfryda — APPROVED** [review]
Looks good to me, thank you @wendycwong !

File: h2o-py/h2o/estimators/model_selection.py (hunk 4 above)
Comment (wendycwong) [reply h2o-py/h2o/estimators/model_selection.py]: Thank you tomas.  Fixed.
